bin/deepstate/main_evm.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+# Copyright (c) 2017 Trail of Bits, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+logging.basicConfig()
+
+import collections
+import sys
+import os
+import md5
+
+try:
+  import manticore.ethereum
+except Exception as e:
+  if "Z3NotFoundError" in repr(type(e)):  
+    print "Manticore requires Z3 to be installed."
+    sys.exit(255)
+  else:
+    raise
+import traceback
+from .common import DeepState
+
+L = logging.getLogger("deepstate.evm")
+L.setLevel(logging.INFO)
+
+def move_tests(workspace, outdir):
+  """Parse and sort testcases"""  
+  tx_files = []
+  for prefix,d,files in list(os.walk(workspace)):
+    for f in files: 
+      if "test" in f and ".tx" in f:
+        tx_files.append(prefix+"/"+ "/".join(d) + f)
+
+  for filename in tx_files:
+      txs = open(filename, "r+").read()
+      test_name = md5.new(txs).hexdigest()
+      if "THROW" in txs:
+          test_name = test_name + ".fail"
+      if "STOP" in txs:
+          test_name = test_name + ".pass" 
+
+      test_file = os.path.join(outdir, test_name)
+      L.info("Saving input to {}".format(test_file))
+      try:
+        with open(test_file, "wb") as f:
+          f.write(txs)
+      except:
+        L.critical("Error saving input to {}".format(test_file))
+
+def compile_contract(contract):
+  """Compile a contract"""
+  m = manticore.ethereum.ManticoreEVM()
+  source_code = open(contract,"r").read()
+  owner_account = m.create_account(balance=1000)
+  contract_account = m.solidity_create_contract(source_code, owner=owner_account, contract_name="TEST")
+  return m, owner_account, contract_account
+ 
+def do_run_test(args, contract, test):
+  """Run an individual test case."""
+  test_name, test_args = test
+  m, owner_account, contract_account = compile_contract(contract)
+  m.verbosity(1)
+  func = contract_account.__getattribute__(test_name)
+  xs = [None]*len(test_args)
+  func(*xs)
+  m.finalize()
```

Comment:
Given that you wrap `do_run_test` with a `try` and `except`, I'm inclined to think that `m.finalize()` should be executed within a `finally` block.